internal/pkg/pluginmanager/manager.go

```diff
@@ -25,37 +30,48 @@ func DownloadPlugins(conf *configloader.Config) error {
 
 	for _, tool := range conf.Tools {
 		pluginFileName := configloader.GetPluginFileName(&tool)
+		pluginMD5FileName := configloader.GetPluginMD5FileName(&tool)
+		// plugin does not exist
 		if _, err := os.Stat(filepath.Join(pluginDir, pluginFileName)); errors.Is(err, os.ErrNotExist) {
-			// plugin does not exist
-			err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version)
-			if err != nil {
+			// download .so file
+			if err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version); err != nil {
 				return err
 			}
-			continue
+			log.Successf("[%s] download succeeded.", pluginMD5FileName)
 		}
-		// check md5
-		dup, err := checkFileMD5(filepath.Join(pluginDir, pluginFileName), dc, pluginFileName, tool.Plugin.Version)
+		// .md5 does not exist
+		if _, err := os.Stat(filepath.Join(pluginDir, pluginFileName)); errors.Is(err, os.ErrNotExist) {
```

Comment:
pluginFileName here should be pluginMD5FileName?

Reply:
updated